### ISSUE TITLE
util.url doesn't work for Beyoncé etc. (utf encoding)

### ISSUE THREAD

**Issue by johnelliott** [cloudinary/cloudinary_npm#147]
The URL function is giving us issues with image file names where there are UTF characters like cedilla and grave accents.

**Problem:** 
The current function results in `Beyonc%327e%301_kofryc`, where the `c` in `Beyonc` is untouched. This URIDecodes to `Beyonc27e01_kofryc`, which isn't the correct public_id.

The desired end url should contain a public_id something like`Beyonc%CC%A7e%CC%81_kofryc` which you can URIDecode back into the original public_id.

**Steps to reproduce:**
Try with a `public_id` like Beyonçé, and stop the debugger as it's looping over the characters in the string: https://github.com/cloudinary/cloudinary_npm/blob/master/src/utils.coffee#L598

It seems that because the regex doesn't match the `c` of the `ç` we're trying to represent, by the time you process the cedilla, the chance for encoding the `c` and the cedilla together into `%CC%A7` has passed:
![image](https://cloud.githubusercontent.com/assets/5633265/25590679/85bd75c4-2e7f-11e7-8515-a31db6a32729.png)

Normalizing the string before sending it to the function doesn't appear to fix the issue.

Helpful links about unicode normalization:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize
http://www.unicode.org/reports/tr15/tr15-29.html

**Comment by taragano** [comment]
Hi @johnelliott,

Thank you for reporting about this.

It seems that there are two different characters that are used here (although visually they look identical), one of them is handled correctly, the other isn't:

```
> cloudinary.utils.url("Beyonçé")     // (Beyon%C3%A7%C3%A9)
'http://res.cloudinary.com/demo/image/upload/Beyon%C3%A7%C3%A9'
> cloudinary.utils.url("Beyon%C3%A7%C3%A9")
'http://res.cloudinary.com/demo/image/upload/Beyon%C3%A7%C3%A9'

> cloudinary.utils.url("Beyonçé")      // (Beyonc%CC%A7e%CC%81)
'http://res.cloudinary.com/demo/image/upload/Beyonc%CC%A7e%CC%81'
> cloudinary.utils.url("Beyonc%CC%A7e%CC%81")
'http://res.cloudinary.com/demo/image/upload/Beyonc%CC%A7e%CC%81'
```
I've passed it on to our team to have a look at it.

**Comment by nadavofi** [comment]
Hi @johnelliott,

Could you please make sure you're using the latest release of this library?
A similar issue has been fixed a while ago (see: #138) with the release of [v1.9.0](https://github.com/cloudinary/cloudinary_npm/releases/tag/1.9.0)

I look forward to your updates.

**Comment by johnelliott** [comment]
@nadavofi Thank you :) We'll compare this code and see if that helps us out.
